classes/check/presigned_urls.php

```diff
@@ -0,0 +1,272 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace tool_objectfs\check;
+
+use action_link;
+use core\check\check;
+use core\check\result;
+use moodle_url;
+use stored_file;
+use tool_objectfs\local\manager;
+
+/**
+ * Test that requested pre-signed URLs respond with the expected file headers.
+ *
+ * @package    tool_objectfs
+ * @author     Andrew Madden <andrewmadden@catalyst-au.net>
+ * @copyright  2022 Catalyst IT
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class presigned_urls extends check {
+
+    /**
+     * A link to a place to action this
+     *
+     * @return action_link
+     */
+    public function get_action_link(): ?\action_link {
+        return new \action_link(
+            new \moodle_url('/admin/settings.php?section=tool_objectfs_settings'),
+            get_string('pluginsettings', 'tool_objectfs'));
+    }
+
+    /**
```

Comment:
Add a details page which just dumps out the list of plugin file urls for each fixture so you can manually retry each one